### PR TITLE
Add support for nested types

### PR DESCRIPTION
**Pull request by ibsh**
Hey Theo, this may be a horrifying oversimplification but essentially allows for type names to be valid even when dot delimited.

### CODE REVIEW

File: Tests/WeaverCodeGenTests/SwiftGenerator/Output/Weaver.test_nested_type.swift [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/// This file is generated by Weaver 0.12.4
+/// DO NOT EDIT!
+// MARK: - FooTest21
+protocol FooTest21DependencyResolver {
+    var foo: Test21Nest.Test21Type<.Test21Type> { get }
```
Comment (LorienMan) [inline Tests/WeaverCodeGenTests/SwiftGenerator/Output/Weaver.test_nested_type.swift]: Hello!
according to 
``` // weaver: foo <= Test21Nest.Test21Type ```
this line should be generated as
``` var foo: Test21Nest.Test21Type { get } ```

Comment (ibsh) [reply Tests/WeaverCodeGenTests/SwiftGenerator/Output/Weaver.test_nested_type.swift]: Well spotted. I’ll take another look.

Comment (LorienMan) [reply Tests/WeaverCodeGenTests/SwiftGenerator/Output/Weaver.test_nested_type.swift]: You can also look through https://github.com/scribd/Weaver/pull/111 , I've faced such a problem too.

Comment (ibsh) [reply Tests/WeaverCodeGenTests/SwiftGenerator/Output/Weaver.test_nested_type.swift]: Oh man, I’m sorry. I didn’t even see your pull request. I’ll close this one, yours is obviously more thorough. I’ll try to find time to review it just for a second pair of eyes.

**Pull request closed without merging**